### DISCUSSION TITLE
feature: client asset hot reloading

### DISCUSSION THREAD

**Discussion by pandinocoder** [AscensionGameDev/Intersect-Engine#2711] [Ideas]
**Is your feature request related to a problem? Please describe.**
In conjunction with #1136 and #1137 it would be strongly beneficial to be able to modify textures and other assets and have Intersect reload them automatically instead of having to restart the client every time a change is made.

**Describe the solution you'd like**
All textures, audio tracks, fonts and shaders should be hot-reloadable, and the client should be crash tolerant.

**Additional context**
This feature should be build-toggleable with the compile flag `INTERSECT_HOT_RELOADING`, so that developers can distribute games with this feature compiled out if they so choose.

**Comment by WeylonSantana** [comment]
thank god and you too
